skglm/estimators.py

```diff
@@ -26,6 +27,123 @@
+        model.classes_ = enc.classes_
+        n_classes_ = len(model.classes_)
+        sparse_format = "csc" if isinstance(datafit, Logistic) else "csr"
+        is_sparse = issparse(X)
+        if n_classes_ <= 2:
+            y = 2 * y - 1
+        X = check_array(
+            X,
+            accept_sparse=sparse_format,
+            dtype=np.float64)
+
+        y = check_array(y, ensure_2d=False, dtype=X.dtype.type)
+        check_consistent_length(X, y)
+    else:
+        check_X_params = dict(
+            dtype=[np.float64, np.float32], order='F',
+            accept_sparse='csc', copy=model.fit_intercept)
+        check_y_params = dict(ensure_2d=False, order='F')
+
+        X, y = model._validate_data(
+            X, y, validate_separately=(check_X_params, check_y_params))
+        X = check_array(X, 'csc', dtype=[np.float64, np.float32],
+                        order='F', copy=False, accept_large_sparse=False)
+        y = check_array(y, 'csc', dtype=X.dtype.type, order='F', copy=False,
+                        ensure_2d=False)
+    if y.ndim == 2 and y.shape[1] == 1:
+        warnings.warn("DataConversionWarning('A column-vector y"
+                      " was passed when a 1d array was expected")
+        y = y[:, 0]
+
+    if not hasattr(model, "n_features_in_"):
+        model.n_features_in_ = X.shape[1]
+
+    n_samples = X.shape[0]
+    if n_samples != y.shape[0]:
+        raise ValueError("X and y have inconsistent dimensions (%d != %d)"
+                         % (n_samples, y.shape[0]))
+
+    if not model.warm_start or not hasattr(model, "coef_"):
+        model.coef_ = None
+
+    penalty = compiled_clone(penalty)
+    datafit = compiled_clone(datafit, to_float32=X.dtype == np.float32)
+    if issparse(X):
+        datafit.initialize_sparse(X.data, X.indptr, X.indices, y)
+    else:
+        datafit.initialize(X, y)
+
+    if isinstance(datafit, QuadraticSVC):
+        if issparse:
+            yXT = (X.T).multiply(y)
+        else:
+            yXT = (X * y[:, None]).T
+        X_ = yXT
+    else:
+        X_ = X
+
+    if model.warm_start and hasattr(model, 'coef_'):
+        w = model.coef_.copy()
+        Xw = X_ @ w
+    else:
+        w = np.zeros(X_.shape[1], dtype=X_.dtype)
+        Xw = np.zeros(X_.shape[0], dtype=X_.dtype)
+
+    # check consistency of weights for WeightedL1
+    if isinstance(penalty, WeightedL1):
+        if len(penalty.weights) != X.shape[1]:
+            raise ValueError(
+                "The size of the WeightedL1 penalty should be n_features"
+                ", expected %i" % X_.shape[1] +
+                " got %i" % len(penalty.weights))
+
+    coefs, p_obj, kkt = cd_solver(
+        X_, y, datafit, penalty, w, Xw, max_iter=model.max_iter,
+        max_epochs=model.max_epochs, p0=model.p0,
+        tol=model.tol, use_acc=True, K=5, ws_strategy=model.ws_strategy,
+        verbose=model.verbose)
+
+    model.coef_, model.stop_crit_ = coefs, kkt
+    model.n_iter_ = len(p_obj)
```

Comment:
add a comment: n_iter for us means number of subproblems (for sklearn it's number of CD epochs, not the same)

Reply:
it is already in the doc of estimators: "max_iter : int, optional The maximum number of iterations (subproblem definitions).", do you want to make a recall here?

Reply:
Nope you are right, I forgot it was there